test/regression/targets/sc-24079.cc

```diff
@@ -0,0 +1,59 @@
+#include <string>
+#include <limits>
+
+#include <tiledb/tiledb>
+
+#include <test/support/tdb_catch.h>
+
+std::string array_name = "cpp_unit_array_24079";
+
+TEST_CASE("C++ API: DoubleDelta filter typecheck should account for output type of FloatScaleFilter", "[cppapi][filter][float-scaling]") {
+    tiledb::Context ctx;
+    tiledb::VFS vfs(ctx);
+
+    if (vfs.is_dir(array_name)) {
+        vfs.remove_dir(array_name);
+    }
+
+    tiledb::Domain domain(ctx);
+    float domain_lo = static_cast<float>(std::numeric_limits<int64_t>::min());
+    float domain_hi = static_cast<float>(std::numeric_limits<int64_t>::max() - 1);
+
+    // Create and initialize dimension.
+    auto d1 = tiledb::Dimension::create<float>(
+        ctx, "soma_joinid", {{domain_lo, domain_hi}}, 2048);
+    
+    tiledb::Filter float_scale(ctx, TILEDB_FILTER_SCALE_FLOAT);
+    double scale = 1.0f;
+    double offset = 0.0f;
+    uint64_t byte_width = 8;
+
+    float_scale.set_option(TILEDB_SCALE_FLOAT_BYTEWIDTH, &byte_width);
+    float_scale.set_option(TILEDB_SCALE_FLOAT_FACTOR, &scale);
+    float_scale.set_option(TILEDB_SCALE_FLOAT_OFFSET, &offset);
+
+    tiledb::Filter dd(ctx, TILEDB_FILTER_DOUBLE_DELTA);
+
+    tiledb::FilterList filters(ctx);
+    filters.add_filter(float_scale);
+    filters.add_filter(dd);
+    d1.set_filter_list(filters);
+    
+    domain.add_dimension(d1);
+
+    auto a = tiledb::Attribute::create<float>(ctx, "A");
+
+    tiledb::ArraySchema schema(ctx, TILEDB_SPARSE);
+    schema.set_domain(domain);
+    schema.add_attribute(a);
+    schema.set_capacity(100000);
+    schema.set_cell_order(TILEDB_ROW_MAJOR);
+    schema.set_tile_order(TILEDB_ROW_MAJOR);
+    CHECK_NOTHROW(tiledb::Array::create(array_name, schema));
```

Comment:
Can you please try to write something to the array after creation ?
I believe your fix indeed suppresses the error on array creation but doesn't fix the actual underlying issue which is that `DoubleDelta` filter needs to somehow _become aware_ that after applying `FloatScale` filter the type will be `integer`. The reason I think that is this part of the code:
```
Status DoubleDelta::compress(
    Datatype type, ConstBuffer* input_buffer, Buffer* output_buffer) {
  switch (type) {
...
    case Datatype::FLOAT32:
    case Datatype::FLOAT64:
      return LOG_STATUS(Status_CompressionError(
          "Cannot decompress tile with DoubleDelta; Float "
          "datatypes are not supported"));
```
where input argument type comes as `auto type = tile.type()` from `CompressionFilter::compress_part`.

Can you try to test it and double check if I got that right?

Reply:
Is the real fix to change the tile's type in the float scale filter? I do think that this is possible. (and by the way, you are correct).

Reply:
I guess this depends on how generic we would like the solution to be. Is `DoubleDelta` the only filter that can/we want to support to be applied after `FloatScale` or we want a more general solution for any filter that comes after it in the pipeline? maybe @ihnorton can give us his opinion on the scope of the issue and solution that we need here?

FYI I have already expressed my concerns on the real problem of our current design of filter pipeline that is that we don't communicate the change in the underlying data type to the next filter in line here: https://app.shortcut.com/tiledb-inc/story/24051/czi-failed-compressing-with-rle-invalid-input-buffer-format .